concrete/src/Mail/Service.php

```diff
@@ -532,4 +617,97 @@ public static function getMailerObject()
             'transport' => $app->make(TransportInterface::class),
         ];
     }
+
+    /**
+     * Convert a list of email addresses to a string.
+     *
+     * @param array $arr
+     *
+     * @return string
+     */
+    protected function generateEmailStrings($arr)
+    {
+        $str = '';
+        for ($i = 0; $i < count($arr); ++$i) {
+            $v = $arr[$i];
+            if (isset($v[1])) {
+                $str .= '"' . $v[1] . '" <' . $v[0] . '>';
+            } elseif (isset($v[0])) {
+                $str .= $v[0];
+            }
+            if (($i + 1) < count($arr)) {
+                $str .= ', ';
+            }
+        }
+
+        return $str;
+    }
+
+    /**
+     * Get the MIME part for the plain text body (if available).
+     *
+     * @return MimePart|null
+     */
+    private function buildTextPart()
```

Comment:
Should these be protected so that this and the sendmail method can be overridden completely?

Reply:
If they are protected we aren't free to change them in future. I think that if someone wants to customize something, they could override the  `sendMail` method...

Reply:
[this comment](https://github.com/concrete5/concrete5/pull/6001#issuecomment-337061421) rings true for these added methods too. 

I think we should try to be a little more pragmatic with how we preserve BC for added properties and methods [like symfony does](https://symfony.com/doc/current/contributing/code/bc.html) (Check out where the bc is not maintained)

Reply:
> I think that if someone wants to customize something, they could override the sendMail method...

The issue I see is that the sendmail method calls those private functions so you'd need to reimplement that functionality in order to override the sendmail method. I'll concede to your judgement on that.

Reply:
Is https://github.com/concrete5/concrete5/pull/6001#issuecomment-337744255 enough pragmatic? :wink:

Reply:
About the `private` methods: I'll mark them as `protected` :+1: